File: qc-core/src/main/java/com/eager/questioncloud/core/domain/point/model/UserPoint.java
package com.eager.questioncloud.core.domain.point.model;

import lombok.Builder;
import lombok.Getter;

@Getter
public class UserPoint {
    private Long userId;
    private int point;

    @Builder
    public UserPoint(Long userId, int point) {
        this.userId = userId;
        this.point = point;
    }

    public void charge(int amount) {
        this.point += amount;
    }

    public void use(int amount) {
        this.point -= amount;
    }
}

File: qc-domain/qc-point/qc-point-core/src/main/kotlin/com/eager/questioncloud/point/domain/UserPoint.kt
package com.eager.questioncloud.point.domain

class UserPoint(
    val userId: Long,
    var point: Int = 0
) {
    companion object {
        fun create(userId: Long): UserPoint {
            return UserPoint(userId = userId)
        }
    }

    fun charge(amount: Int) {
        this.point += amount
    }

    fun use(amount: Int) {
        this.point -= amount
    }
}
